README.md

```diff
@@ -73,23 +73,32 @@ Replace `<username>` with your LPC username. Then `cd` to the directory of your
 git clone https://github.com/FNALLPC/machine-learning-hats
 ```
 
-In order to open Jupyter, you will need to have either installed Jupyter in your user are or have a CMSSW environment setup. I usually choose any CMSSW directory I've already setup, just for this initial step. In the future, we will have installed CMSSW inside the tutorial folder.
+In order to open Jupyter with all the appopriate libraries, you will need to have either installed a `conda` environment Jupyter in your user are or have a CMSSW environment setup. To install and activate a `conda` environment you can do:
+```bash
+wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O $HOME/miniconda3.sh
+bash $HOME/miniconda3.sh -b -f -u -p $HOME/miniconda3
+source $HOME/miniconda3/etc/profile.d/conda.sh
+conda env create -f environment.yml --name machine-learning-hats-2020
+conda activate machine-learning-hats-2020
+```
 
-Once you have a CMSSW environment, open Jupyter:
+Once you have a conda environment, open Jupyter:
 ```bash
 jupyter notebook --no-browser --port=8888 --ip 127.0.0.1
 ```
 If everything worked, the last line of the output should be a url of the form:
 ```bash
 http://127.0.0.1:8888/?token=<long string of numbers and letters>
 ```
-Copy this url into your browser. You may now perform the rest of the exercise like normal, begining by running notebook [`0-setup-libraries.ipynb`](0-setup-libraries.ipynb).
+Copy this url into your browser. You may now perform the rest of the exercise like normal.
```

Comment:
How do they then get the Jupyter kernel? Won't they need to run notebook 0-setup-libraries.ipynb? If so, won't it overwrite, or try to overwrite, the original miniconda distribution? You might want to put if statements around the `wget`, `bash`, environment creation, wrapper creation, and kernel creation lines to see if these already exist and if they do, skip that step. For example:
```bash
if [[ ! -f $HOME/miniconda3.sh ]]; then
    wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O $HOME/miniconda3.sh
fi
if [[ ! -d $HOME/miniconda3 ]]; then
    bash $HOME/miniconda3.sh -b -f -u -p $HOME/miniconda3
    source $HOME/miniconda3/etc/profile.d/conda.sh
    conda env create -f environment.yml --name $KERNEL_NAME
else
    source $HOME/miniconda3/etc/profile.d/conda.sh
fi
conda activate $KERNEL_NAME

if [[ ! -f python_wrapper.sh ]]; then
    <making the python wrapper part>
fi

kgrep=`jupyter kernelspec list | grep machine-learning-hats-2020`
rc=$?
if [[ "$rc" != "0" ]]; then
    <kernel creation part>
fi
```

This is going to need a fair amount of testing. The cases I see are:
1. Initial setup
2. Someone tries to run the setup a second time and nothing should happen. Maybe print some default messages saying things are already setup/exist.
3. Someone has miniconda because they followed the directions above, but are now trying to install the kernel. In this case the conda setup part should output some messages about miniconda already being there, but then the kernel creation part should proceed like normal.

Reply:
BTW, each person will now have a 6.7 GB miniconda distribution in their home area. This will work for ACCRE, but not the LPC, which only provides 2.0 GB to the home area. You might want to put a note about switching the miniconda install location if working on the LPC, or have a switch that does this automatically. For example, instead of using the $HOME environment variable use:
```bash
if [[ `hostname` == *"cmslpc"*".fnal.gov"* ]]; then
    INSTALLATION_DIR=${HOME}/nobackup 
else
    INSTALLATION_DIR=${HOME}
fi
```

Reply:
I tested the cases you pointed out on Vanderbilt and it seemed to work. I also added your suggestions to the notebook.

Regarding LPC, I tried to clarify in the README: when you create your own conda environment and launch Jupyter *from that conda environment*, you don't need any special kernel (the default kernel contains all the libraries in that conda environment).